utils/unittest/chain_suite.go

```diff
@@ -515,18 +519,23 @@ func (bc *BaseChainSuite) ValidSubgraphFixture() subgraphFixture {
 
 func (bc *BaseChainSuite) Extend(block *flow.Block) {
 	bc.Blocks[block.ID()] = block
-	bc.SealsIndex[block.ID()] = bc.SealsIndex[block.Header.ParentID]
+	if seal, ok := bc.SealsIndex[block.Header.ParentID]; ok {
+		bc.SealsIndex[block.ID()] = seal
+	}
+
+	resultsByID := block.Payload.ResultsById()
 
 	for _, receipt := range block.Payload.Receipts {
+		result := resultsByID[receipt.ResultID]
```

Comment:
Can we change this to (?)
```suggestion
	for _, result := range block.Payload.Results {
```
Theoretically, verification is focused only on results (not receipts). If this is possible without lots of test failures, it would be great to get it done as part of this PR. 

If this causes a lot of test failure, I am happy to postpone this update to a subsequent PR (in this case, could you please create an issue? thanks).

Reply:
Applied this change.